holidays/countries/paraguay.py

```diff
@@ -154,46 +156,45 @@ def _populate(self, year):
             2018: (FEB, 26),
             2022: (FEB, 28),
         }
-        self[
-            date(year, *dates_obs.get(year, (MAR, 1)))
-        ] = "Día de los Héroes de la Patria [Patriots Day]"
+        self._add_holiday(
+            "Día de los Héroes de la Patria [Patriots Day]",
+            *dates_obs.get(year, (MAR, 1)),
+        )
```

Comment:
To keep the original logic, here we need to use `super()._add_holiday()`, because `self._add_holiday()` will ignore holiday when it falls on weekend and `self.observed == False`. Or not override `_add_holiday()`, but add a separate function.
Similarly, in other cases where `self[...] = ...` is used originally.

Reply:
Reverted the PY changes for now. I just want to get the #865 finally done 😭

Reply:
I'm waiting for it too! ✌️